create-rust-app/src/util/workspace_utils.rs

```diff
@@ -0,0 +1,84 @@
+use std::path::{Path, PathBuf};
+
+use lazy_static::lazy_static;
+
+lazy_static!(
+    pub static ref WORKSPACE_DIR: PathBuf = {
+        let output = std::process::Command::new(env!("CARGO"))
+            .arg("locate-project")
+            .arg("--workspace")
+            .arg("--message-format=plain")
+            .output()
+            .unwrap()
+            .stdout;
+        let cargo_path = Path::new(std::str::from_utf8(&output).unwrap().trim());
+        cargo_path.parent().unwrap().to_path_buf()
+        // .to_str()
+        // .unwrap()
+        // .to_owned()
+    };
```

Comment:
the rest looks good to me but here we're assuming we'll have the project's source code. That's not necessarily the case (see #332 -- we split the build and binary container into two). We also don't want to go in this direction as it stops us from attempting a "single binary" build in the future as well.

there's some tech debt here that needs to be addressed

Reply:
Should we have it stored in an environment variable instead? Similar to frontenddir

It's just that we need a way to know where the base of the project is and that's not always the current working directory

Reply:
Yeah. Some more thoughts:

I think we need to remove the feature flag. In `create-rust-app/src/dev/mod.rs`, we should check if it's a workspace using the `Cargo.toml` instead of relying on a feature flag.

Also, all the variables in this file should be `&'static str` and be initialized to some default unless the backing environment variable is present (and they should all be configurable by env vars).

Let me know what you think!

Reply:
I've implemented most of your suggestions as of the most recent commits (also got rid of some unnecessary unwraps), but there's a few things I still need to figure out:

- a function that can tell us if we're using workspaces (without needing source code) (rather than using the feature flag)
- a way of finding the workspace root when the (not yet added) CRA_WORKSPACE_ROOT environment variable isn't set

Reply:
btw, sorry I've been awol for so long, I was busy at work over the summer and now school's picking up again (well actually, the semester is almost over, but still) so I haven't had much time.

Reply:
sounds awesome -- it's cool to hear about your journey so far :)

Reply:
I'll have to take some time out to look into this further

Reply:
I think I figured something out, high level is that I made "finding the workspace dir" infallible (no panics, even in a container or something), then rethought the logic used in initializing these variables (getting rid of the need for the feature flag)

Reply:
after this, I'll take a look at general maintenance and whatnot, but let's get this out of the way first to prevent merge conflicts

Reply:
@Wulf is this resolved?

Reply:
beautifully done.

🙌